python/dgl/contrib/dis_sampling/sender.py

```diff
@@ -0,0 +1,115 @@
+# This code implements a distributed dgl-subgraph sender, which 
+# samples sub-graphs continuously from a huge graph and sends sub-graphs
+# to a reciver node, which uses these sub-graphs for model training.
+import mxnet as mx
+import numpy as np
+
+import time
+
+import grpc
+import sampler_pb2
+import sampler_pb2_grpc
+
+_MAX_MESSAGE_LENGTH = 1024 ** 3  # Gigabyte
+
+def serilize(mx_ndarray):
+    """ This method serilizes a mxnet ndarray to memory.
+    It will first convert mxnet ndarray to numpy ndarray,
+    and then serilize it to binary string.
+
+    Args:
+      mx_ndarray: mxnet ndarray
+    Returns:
+      serialized binary string
+    """
+    np_ndarray = mx_ndarray.asnumpy()
+    str = np_ndarray.tostring()
+    return str
+
+def deserilize(str_array, type):
+    """ This method deserilizes a mxnet ndarray from a binary string.
+    It first serlizes the binary string to a numpy array, and then
+    converts it to a mxnet ndarray.
+
+    Args:
+      str_array: binary ndarray
+      type: data type
+    Returns:
+      mxnet ndarray
+    """
+    np_ndarray = np.frombuffer(str_array, dtype=type)
+    mx_ndarray = mx.nd.array(np_ndarray, dtype=type)
+    return mx_ndarray
+
+class Sender(object):
+    """ Send sub-graphs to reciver
+    """
+    def __init__(self, addr):
+        """ Constructor
+
+        Args:
+            addr: remote address string, e.g., '162.168.8.12:50051'
+        """
+        self.addr = addr
+
+    def bind(self, fn, graph, seeds, num_args, num_hops, num_neighbor, max_num_vertices):
```

Comment:
I don't think we have to pass `graphs`, `seeds`, etc to `bind`. All these information can be carried in the closure of `fn`.

Reply:
These arguments are specific to neighbor sampling.

Reply:
Yes, we can make it more general here.